File: 프로그래머스/lv2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    queue = deque()
    
    for i, p in enumerate(priorities):
        queue.append((i,p))
    
    while len(queue) > 0:
        max_priority = max([q[1] for q in queue])
        process = queue.popleft()
        
        if process[1] < max_priority:
            queue.append(process)
        else:
            answer += 1
            if process[0] == location:
                return answer
        

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<Integer> queue = new LinkedList<>();
        Queue<Integer> queue2 = new LinkedList<>();
        for(int i=0;i<priorities.length;i++) {
            queue.offer(priorities[i]);
            queue2.offer(i);
        }
        while (!queue.isEmpty()) {
            Integer process = queue.poll();
            Integer loc = queue2.poll();
            if (!queue.isEmpty()) {
                if (process >= Collections.max(queue)) {
                    answer += 1;
                    if (loc == location) {
                        return answer;
                    }
                } else {
                    queue.offer(process);
                    queue2.offer(loc);
                }
            } else {
                return answer + 1;
            }
        }
        return answer;
    }
}